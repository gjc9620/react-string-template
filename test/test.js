var format = require("string-template");
var template = require("../index");

console.log(
  format("Hello {{name}, you have {count} {vvv} {{aaa}} unread messages", {
    name: "Robert",
    count: 12
  })
);

template(
  {
    str: "Hello {{name}, you have {count} {vvv} {{aaa}} unread messages",
    values: {
      name: "Robert",
      count: 12
    },
    render: (arr)=>console.log(arr.join('')),
  }
);


///////////////////////////////////////

console.log(
  format("111{name}1111", {
    name: "Robert",
    count: 12
  })
);

template(
  {
    str: "111{name}1111",
    values: {
      name: "Robert",
      count: 12
    },
    render: (arr)=>console.log(arr.join('')),
  }
);


///////////////////////////////////

console.log(
  format("1111111", {
    name: "Robert",
    count: 12
  })
);

template(
  {
    str: "1111111",
    values: {
      name: "Robert",
      count: 12
    },
    render: (arr)=>console.log(arr.join('')),
  }
);


/////////////////////////////////// no match


template(
  {
    str: "111{ggg}1111",
    renderNoMatch: ()=>'AAAA',
    values: {
      name: "Robert",
      count: 12
    },
    render: (arr)=>console.log(arr.join('')),
  }
);

template(
  {
    str: "111{ggg}1111",
    renderNoMatch: ()=>'{}{}{}}{}{{{}}}{AAA}',
    values: {
      name: "Robert",
      count: 12
    },
    render: (arr)=>console.log(arr.join('')),
  }
);


///////////////////////////////////


template(
  {
    str: "111{name}11{count}11{bad}ddddddd{sym}",
    renderNoMatch: ()=>'{}{}{}}{}{{{}}}{AAA}',
    values: {
      name: function () {},
      count: { type: 1},
      bad: undefined,
      sym: Symbol("A"),
    },
    render: (arr)=>console.log(arr),
  }
);
