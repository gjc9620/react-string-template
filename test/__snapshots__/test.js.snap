// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If not match 1`] = `
<div>
  I am 
  not match
  
</div>
`;

exports[`Insert React element and click event 1`] = `
<div>
  
  <span
    onClick={[Function]}
  >
    Patronus
  </span>
   Charm
</div>
`;

exports[`With React element 1`] = `
<div>
  
  <a
    href="https://www.google.com/search?q=Harry Potter"
  >
    Harry Potter
  </a>
   is a series of 
  fantasy
   novels written by British {author} 
  J. K. Rowling
   in 
  1997
  .
</div>
`;
